Minerals.gd

```diff
@@ -0,0 +1,75 @@
+extends Spatial
+
+
+# Declare member variables here. Examples:
+enum {
+	NODE,
+	PASSED_TIME,
+	RESPAWN_TIME,
+	STATE
+}
+
+var scene = null
+var mineral = null
+var listBroken = [] # [[mineral, 0, 30, false]] in it 
+var thread
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	scene = get_tree().get_root()
+	loadStateMinerals()
+	thread = Thread.new()
+	thread.start(self, "respawnMinerals", listBroken)
```

Comment:
If you start a thread you should join the thread when the node is removed from the tree